src/Autocomplete/src/Form/AutocompleteEntityTypeSubscriber.php

```diff
@@ -73,11 +73,15 @@ public function preSubmit(FormEvent $event)
                     ++$idx;
                 }
 
-                $options['choices'] = $repository->createQueryBuilder('o')
-                    ->where(sprintf("o.$idField IN (%s)", implode(', ', array_keys($params))))
-                    ->setParameters(new ArrayCollection($params))
-                    ->getQuery()
-                    ->getResult();
+                $queryBuilder = $repository->createQueryBuilder('o');
+
+                if ($params) {
```

Comment:
Why don't just check if `$data['autocomplete']` is `[]` and skip this else part?
Btw this can be simplified a lot like `$repository->createQueryBuilder('o')->->where(sprintf('o.%s IN (:ids)', $idField)))->setParameter('ids', $data['autocomplete'])->getQuery()->getResult()`, doctrine will handle the rest, doesn't it?

Reply:
No it won't handle the types properly,that's what I've fixed before.